### PR TITLE
[CMS-272] Use kubeadm.k8s.io/v1alpha1 and cleanup before installation.

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

**Comment by joejulian** [comment]
Do we want to wrap this in some sort of logic to we can upgrade versions?

**Comment by alejandroEsc** [comment]
The change is acceptable thought i havent had this issue, even with new prestine deployments. Could you verify that the new apiversion works and consistently deploys?

**Comment by alejandroEsc** [comment]
@davidewatson i think i got your verification on slack. going to merge.

**Comment by joejulian** [comment]
Additionally, v1alpha1 read support exists in v1.11, but will be removed in v1.12: https://github.com/kubernetes/kubernetes/pull/63788

File: clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
           # Set up kubeadm config file to pass parameters to kubeadm init.
           cat << EOF | sudo tee /etc/kubernetes/kubeadm_config.yaml
-          apiVersion: kubeadm.k8s.io/v1alpha2
```
Comment (joejulian) [inline clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template]: Please add a comment line explaining when we will need to switch to v1alpha2.

Comment (alejandroEsc) [reply clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template]: should be backwards compatible? if not yeah big deal. Good point!

**Comment by davidewatson** [comment]
@joejulian: I can add the requested logic. We'll need it later. The purpose of this PR is to ensure everyone is on the same page. I was going to submit another temporary PR along the lines of:

```
          kubeadm reset || true
          apt-get purge kubeadm kubectl kubelet kubernetes-cni kube* -y || true
          rm -f /etc/systemd/system/kubelet.service.d/* || true
          rm -f /etc/systemd/system/kubelet.d/* || true
          rm -rf /var/lib/etcd || true
```

I think this is even worse than the current PR. :) Is it okay if we defer this change for when we rewrite these scripts? The are many problems (e.g. unpinned and untested docker releases, etc.)

**Comment by alejandroEsc** [comment]
@davidewatson I am missing context here. What are you suggestion you do with that kubeadm reset || true? I hope you are suggesting we run that as part of the provisioning script.

**Comment by davidewatson** [comment]
@alejandroEsc: That is correct.

**Comment by alejandroEsc** [comment]
@davidewatson i think that brings forth a lot of assumptions that we should not make. First we assume that machines are Pristine. Making sure that is true or fixing any state issues with existing machines should not be our concern. We should make those assumptions clear and known. As for ourselves and our testing, we should do the cleaning we can and make sure things are pristine before each test cycle. That is on us. Honestly I dont want to support every possible combination of things to look for or clean, or even worry about supporting such code.

File: clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template [hunk 2]
```diff
@@ -6,6 +6,13 @@
           # Run in noninteractive mode to avoid erroring when a config file preexists.
           export DEBIAN_FRONTEND=noninteractive
 
+          # This a temporary hack for development purposes.
```
Comment (alejandroEsc) [inline clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template]: Please DO NOT INCLUDE THIS even with a TODO. Please create a new master template file and call it _DEV if you wish.

Comment (joejulian) [reply clusterctl/examples/ssh/bootstrap_scripts/master_ubuntu_16.04.template]: Even further, this should be in a separate PR. Note you had to use the word `and` in your PR subject.

**Comment by davidewatson** [comment]
@alejandroEsc, @joejulian: I've removed it hack.

Again, the intent of the PR was to simplify things while over half of our team can't seem to deploy. Let's focus on that issue please.

**Pull request merged**